pkg/controller/volume/scheduling/scheduler_binder.go

```diff
@@ -348,10 +348,17 @@ func (b *volumeBinder) BindPodVolumes(assumedPod *v1.Pod) (err error) {
 		return err
 	}
 
-	return wait.Poll(time.Second, b.bindTimeout, func() (bool, error) {
+	if err = wait.Poll(time.Second, b.bindTimeout, func() (bool, error) {
 		b, err := b.checkBindings(assumedPod, bindings, claimsToProvision)
 		return b, err
-	})
+	}); err != nil {
+		pvcName := ""
+		if len(claimsToProvision) > 0 {
+			pvcName = claimsToProvision[0].Name
+		}
+		return fmt.Errorf("selected node of PVC %q does not match the expected node, may be rejected by the provisioner: %v", pvcName, err)
```

Comment:
please understand the code first, it's not correct to add this error message here.

Reply:
no, we should update the internal error message [here](https://github.com/kubernetes/kubernetes/blob/06be9af4a4c531b6c791a4ecc09659973e82b838/pkg/controller/volume/scheduling/scheduler_binder.go#L521).

Reply:
@jsafrane 
sorry, my comment [here](https://github.com/kubernetes/kubernetes/pull/83343#issuecomment-537837685) is misleading, the code path I want to say is the code you quoted [in your issue](https://github.com/kubernetes/kubernetes/issues/83181#issue-498739060).

`BindPodVolumes` may fail a) PV controller fails to bind static PV or b) provisioner fails to provision a volume. we should return "provisioning failed ...` error message when `selectedNode` is removed from a PVC

Reply:
https://github.com/kubernetes/kubernetes/pull/83540